docs/idempotence-patterns.md

```diff
@@ -0,0 +1,3 @@
+# Idempotence Patterns
+
+[ TODO ]
```

Comment:
Still todo?

Reply:
Yep.  We know some patterns, and will learn more in time.  This will end up being a pretty long section, deserving its own page, and I had grown tired of writing docs for now.  So we can flesh it out later.